File: 1018 - Binary Prefix Divisible By 5.py
class Solution:
    def prefixesDivBy5(self, nums):
        res = []
        remain = 0
        for n in nums:
            remain = (remain * 2 + n) % 5
            res.append(remain == 0)
        return res

s = Solution()
a = [1,1,0,0,0,1,0,0,1]
b = s.prefixesDivBy5(a)
print(b)

File: 1018 - Binary Prefix Divisible By 5.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> res = new ArrayList<>();
        int dec = 0;
        for (int n : nums) {
            dec = (dec * 2 + n) % 5;
            res.add(dec == 0);
        }
        return res;
    }
}